unit/java_bytecode/java_bytecode_parse_generics/parse_generic_superclasses.cpp

```diff
@@ -0,0 +1,63 @@
+/*******************************************************************\
+
+ Module: Unit tests for parsing classes with generic superclasses with
+         unsupported signatures, falling back to using the raw type
+         descriptors
+
+ Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+#include <testing-utils/catch.hpp>
+#include <testing-utils/load_java_class.h>
+#include <testing-utils/require_type.h>
+
+SCENARIO(
+  "parse generic superclass signature",
+  "[core][java_byte code[java_bytecode_parse_generics]]")
+{
+  const symbol_tablet &new_symbol_table = load_java_class(
+    "GenericBounds", "./java_bytecode/java_bytecode_parse_generics");
+
+  const std::string base_generic = "java::Generic";
+  const irep_idt base_generic_interface = "java::InterfaceGeneric";
+
+  const std::string load_class("java::GenericBounds");
+  THEN(
+    "these fields have a non-generic base class / interface  as their real "
+    "generic signature is unsupported at the moment")
+  {
+    // once bounds in generic signatures are supported, this test must be
+    // changed to check for the correct generic types
```

Comment:
Please mark this as TODO, you can cite TG-1286 (parsing bounds) and TG-675 (parsing wildcards).